### DISCUSSION TITLE
Xiaomi Door and Windows Sensor 2 connection question.

### DISCUSSION THREAD

**Discussion by dimkagithub** [merdok/homebridge-miot#442] [Q&A] [answered]
Hi!
How can I connect isa.magnet.dw2hl (Xiaomi Door and Windows Sensor 2)?
I have Xiaomi Smart LED Ceiling Light MJXDD01SYL with Bluetooth gateway.

Ceiling light is connected to HomeKit. I dont use native Xiaomi, Mi, etc software. I don't even want to use it further.

I cant understand, should I connect Ceiling light to MiHome app too and then connect sensor. Or I it is possible to connect sensor without MiHome software, just using Homebridge Miot?

Thanks)

**Comment by merdok** [comment, marked as answer]
You need to retrieve the token from your devices, the easiest way is to add them to your micloud account using the mi home app and then let the plugin retrieve the data by specifying your mi cloud account.
Also note that for Bluetooth devices you need to specify the gateway ip and token but the deviceid and model has to be from the Bluetooth device which you are trying to control.

**Reply by dimkagithub** [reply]
Done as you suggested.
It successfully connected to BT Hub.

```
[02.03.2023, 21:54:19] [miot] [Door Sensor] Device found! Setting up miot device from local connection!
[02.03.2023, 21:54:19] [miot] [Door Sensor] Connected to device: yeelink.light.ceiling22
[02.03.2023, 21:54:19] [miot] [Door Sensor] Doing initial property fetch.
[02.03.2023, 21:54:27] [miot] [Door Sensor] Starting property polling.
[02.03.2023, 21:54:33] [miot] [Door Sensor] Poll failed 7 times in a row! Stopping polling and trying to reconnect! Reason: Error: busy.
```

Discover all devices via MiCloud doesn't find Door And Window Sensor. Just vacuum and ceiling light.

Now deviceid property is empty.

How can I get Door & Window sensor deviceID?

**Reply by merdok** [reply]
That I am not quite sure as I do not own a gateway or any Bluetooth devices, but I guess they should definitely also appear in your mi cloud account when you add them in the mi home app.

**Reply by dimkagithub** [reply]
No, it shows Dreame D9, Xiaomi Ceiling Light only.

**Reply by merdok** [reply]
Then I unfortunately do not know any other way on how to retrieve that from ble devices...

**Reply by dimkagithub** [reply]
> Then I unfortunately do not know any other way on how to retrieve that from ble devices...

I solved the problem! Now sensor works fine!
In your plugin "Discover All Devices via miCloud" does not show Bluetooth devices. Therefore I could not get my Door And Window Sensor deviceid.

I used Xiaomi Cloud Tokens Extractor and It showed all devices include Bluetooth ones!
The Xiaomi Bluetooth device id look like: blt.x.xxxxxxxxxxxx

**Reply by merdok** [reply]
Ah, you are right! I filter out bluetooth devices from that list as they were not working initially, forgot about that😅

**Reply by dimkagithub** [reply]
> Ah, you are right! I filter out bluetooth devices from that list as they were not working initially, forgot about that😅

It would be better not to forget to turn off this filter. 😅

**Reply by merdok** [reply]
As of version 1.5.2 it is now possible to show BLE devices in the discover Ui
